### DISCUSSION TITLE
FBP_CUDA ‘ShortScan' option for 'Fanflat' Geometry

### DISCUSSION THREAD

**Discussion by MarkatLiu** [astra-toolbox/astra-toolbox#430] [Q&A]
Hi, currently, I need to use the FBP_CUDA ShortScan option to reconstruct the projection data from the scanner. But it seems that the ShrotScan option only works for the 'FanFlat' geometry. For the 'FanFlat_vec', it doesn't work. Are there any soultions for it? 
Besides, when I try to use the 'FanFlat' geometry instead, I find that there are no interfaces to configure the rotation axis offsets, is it possible to do this for 'FanFlat' geometry? Thanks.

**Comment by askorikov** [comment]
This is because short scan correction was designed for circular geometry, whereas with `fanflat_vec` option we assume that the geometry is more complex and therefore simple short scan correction will not be sufficient. Nevertheless, I we will consider letting the user apply it anyway and only warn about the potential artifacts for non-circular geometries.

For your second question, do you mean that the function for correcting the rotation axis offset throws an error with the fanflat geometry? If so, it should get fixed by https://github.com/astra-toolbox/astra-toolbox/commit/9d0cf97ebd465a0046db852c48678cdccd9cccdb in the next release.

**Reply by MarkatLiu** [reply]
Hi askorikov, for the second question, the problem to me is that there seems to be no methods to correct the rotation axis offset for the fanflat geometry. Is my understanding correct?

**Reply by askorikov** [reply]
Well, I think the same `geom_postalignment` function should work for correcting the rotation axis offset for both parallel and fanflat/cone geometries, unless I misunderstand your situation.

**Reply by MarkatLiu** [reply]
Hi askorikov, the detailed problem I currently have is that the 'ShortScan' option can be configured correctly without any warning with 'falflat' when the geom_postalignment is not called to correct the rotation axis offset. However, when the geom_postalignment is called, the following warning will be reported which indicates the 'ShortScan' doesn't take effect.
'Warning: CudaFilteredBackProjectionAlgorithm: unused configuration options: ShortScan'

**Reply by askorikov** [reply]
I see. Then it's what I meant in the first paragraph of the original reply: if geometry is of `fanflat_vec` type, we assume that a simple short scan correction is not valid. But I understand that you what to apply it anyway even though your geometry is not circular?
